srtcore/core.cpp

```diff
@@ -9525,65 +9525,75 @@ bool srt::CUDT::packUniqueData(CPacket& w_packet, time_point& w_origintime)
     // when scheduling, not when extracting. This is a inter-migration form,
-    // so still override the value, but trace it.
+    // only override extraction sequence with scheduling sequence in group mode.
     m_iSndCurrSeqNo = CSeqNo::incseq(m_iSndCurrSeqNo);
 
-    // Do this checking only for groups and only at the very first moment,
-    // when there's still nothing in the buffer. Otherwise there will be
-    // a serious data discrepancy between the agent and the peer.
-    // After increasing by 1, but being previously set as ISN-1, this should be == ISN,
-    // if this is the very first packet to send.
 #if ENABLE_EXPERIMENTAL_BONDING
-    // Fortunately here is only the procedure that verifies if the extraction
-    // sequence is moved due to the difference between ISN caught during the existing
-    // transmission and the first sequence possible to be used at the first sending
-    // instruction. The group itself isn't being accessed.
-    if (m_parent->m_GroupOf && m_iSndCurrSeqNo != w_packet.m_iSeqNo && m_iSndCurrSeqNo == m_iISN)
-    {
-        const int packetspan = CSeqNo::seqcmp(w_packet.m_iSeqNo, m_iSndCurrSeqNo);
-
-        HLOGC(qslog.Debug, log << CONID() << "packData: Fixing EXTRACTION sequence " << m_iSndCurrSeqNo
-                << " from SCHEDULING sequence " << w_packet.m_iSeqNo
-                << " DIFF: " << packetspan << " STAMP:" << BufferStamp(w_packet.m_pcData, w_packet.getLength()));
-
-        // This is the very first packet to be sent; so there's nothing in
-        // the sending buffer yet, and therefore we are in a situation as just
-        // after connection. No packets in the buffer, no packets are sent,
-        // no ACK to be awaited. We can screw up all the variables that are
-        // initialized from ISN just after connection.
-        //
-        // Additionally send the drop request to the peer so that it
-        // won't stupidly request the packets to be retransmitted.
-        // Don't do it if the difference isn't positive or exceeds the threshold.
-        if (packetspan > 0)
-        {
-            int32_t seqpair[2];
-            seqpair[0] = m_iSndCurrSeqNo;
-            seqpair[1] = w_packet.m_iSeqNo;
-            HLOGC(qslog.Debug, log << "... sending INITIAL DROP (ISN FIX): "
-                    << "msg=" << MSGNO_SEQ::unwrap(w_packet.m_iMsgNo) << " SEQ:"
-                    << seqpair[0] << " - " << seqpair[1] << "(" << packetspan << " packets)");
-            sendCtrl(UMSG_DROPREQ, &w_packet.m_iMsgNo, seqpair, sizeof(seqpair));
+    // Fortunately the group itself isn't being accessed.
+    if (m_parent->m_GroupOf)
+    {
+        if (m_iSndCurrSeqNo != w_packet.m_iSeqNo)
+        {
+            const int packetspan = CSeqNo::seqcmp(w_packet.m_iSeqNo, m_iSndCurrSeqNo);
+            // After increasing by 1, but being previously set as ISN-1, this should be == ISN,
+            // if this is the very first packet to send.
+            if (m_iSndCurrSeqNo == m_iISN)
```

Comment:
It is hard to imagine the reason, but what if `(m_iSndCurrSeqNo == m_iISN && packetspan < 0)`?
It should be unexpected and also an IPE. But in this case, the socket would be fixed behind the ISE. 🤔